components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -656,26 +655,32 @@ void JsonRpcService::UpdateIsEip1559(const std::string& chain_id,
   }
 }
 
-void JsonRpcService::FireNetworkChanged(mojom::CoinType coin) {
+void JsonRpcService::FireNetworkChanged(
+    mojom::CoinType coin,
+    const absl::optional<url::Origin>& origin) {
   for (const auto& observer : observers_) {
-    observer->ChainChangedEvent(GetChainId(coin), coin);
+    observer->ChainChangedEvent(GetChainId(coin, origin), coin, origin);
   }
 }
 
-std::string JsonRpcService::GetChainId(mojom::CoinType coin) const {
-  return chain_ids_.contains(coin) ? chain_ids_.at(coin) : std::string();
+std::string JsonRpcService::GetChainId(
+    mojom::CoinType coin,
+    const absl::optional<::url::Origin>& origin) const {
+  return GetCurrentChainId(prefs_, coin, origin);
 }
 
 void JsonRpcService::GetChainId(
     mojom::CoinType coin,
+    const absl::optional<::url::Origin>& origin,
     mojom::JsonRpcService::GetChainIdCallback callback) {
-  std::move(callback).Run(GetChainId(coin));
+  std::move(callback).Run(GetChainId(coin, origin));
 }
```

Comment:
1) I'd suggest adding Sync suffix to the first one
2) Also I'd suggest having two methods `GetDefaultChainId(CoinType coin)` and `GetChainIdForOrigin(CoinType coin, url::Origin origin)`

Reply:
fixed in a5d2e521422403c3d6e9572dfec0e054afdcef79